src/components/overlay/index.js

```diff
@@ -0,0 +1,94 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {StyleSheet, Image} from 'react-native';
+import {Colors} from '../../style';
+import {PureBaseComponent} from '../../commons';
+import View from '../view';
+
+
+const gradientImage = require('./assets/GradientOverlay.png');
+const OVERLY_TYPES = {
+  DEFAULT: 'default',
+  TOP: 'top',
+  BOTTOM: 'bottom',
+  SOLID: 'solid'
+}
+/**
+ * @description: Overlay view with types (default, top, bottom, solid)
+ * @extends: Image
+ * @extendsLink: https://facebook.github.io/react-native/docs/image
+ */
+export default class Overlay extends PureBaseComponent {
+  static displayName = 'Overlay';
+
+  static propTypes = {
+    /** 
+    * The type of overlay to set on top of the image 
+    */
+    type: PropTypes.oneOf(_.values(OVERLY_TYPES))
+  };
+
+  static overlayTypes = OVERLY_TYPES;
+
+  getStyleByType() {
+    switch (this.props.type) {
+      case OVERLY_TYPES.TOP: 
+        return styles.top;
+      case OVERLY_TYPES.BOTTOM: 
+        return styles.bottom;
+      case OVERLY_TYPES.SOLID: 
+        return styles.solid
+      default:
+        break;
+    }
+  }
+
+  renderImage(typeStyle) {
+    const {type, style} = this.props;
```

Comment:
if someone passed `tintColor` (for example) inside the style for the actual image, wouldn't it be passed to the gradient of the overlay too?

Reply:
So what are you suggesting? I need the layout style from the image